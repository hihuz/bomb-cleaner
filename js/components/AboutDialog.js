import React from 'react';

const AboutDialog = ({ closeAboutDialog, opened }) => (
  <div className={`dialog${opened?' opened':''}`}>
    <button className='dialog-close' onClick={closeAboutDialog}>
      <i className='icon-close'></i>
    </button>
    <p>
      This is a MineSweeper game implemented with React and Redux.
      <br />
      <br />
      You can play it in the same way as the original MineSweeper.
    </p>
    <ul>
      <li>Left Click to open a tile</li>
      <li>Right Click to "flag" a tile</li>
      <li>You'll lose if you open a mine</li>
      <li>You'll win if you open all mine-free tiles</li>
    </ul>
    <p className='centered-text'>
      Good luck !
      <br />
      <br />
      <a target="_blank" href="https://twitter.com/hihuz">@hihuz</a>
    </p>
  </div>
);
export default AboutDialog;
