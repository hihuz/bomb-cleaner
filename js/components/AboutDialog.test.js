import React from 'react';
import AboutDialog from './AboutDialog';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Snapshot', () => {
  const component = shallow(<AboutDialog />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should call the closeDialog prop when the dialog-close button is clicked', () => {
  const clickFn = jest.fn();
  const component = shallow(<AboutDialog closeDialog={clickFn}/>);
  component.find('button').simulate('click');
  expect(clickFn).toHaveBeenCalledTimes(1);
});
