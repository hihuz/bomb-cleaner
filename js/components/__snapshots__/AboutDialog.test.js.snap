exports[`test Snapshot 1`] = `
<div>
  <button
    className="dialog-close">
    <i
      className="icon-close" />
  </button>
  <p>
    This is a MineSweeper game implemented with React and Redux.
    <br />
    <br />
    You can play it in the same way as the original MineSweeper.
  </p>
  <ul>
    <li>
      Left Click to open a tile
    </li>
    <li>
      Right Click to \"flag\" a tile
    </li>
    <li>
      You\'ll lose if you open a mine
    </li>
    <li>
      You\'ll win if you open all mine-free tiles
    </li>
  </ul>
  <p
    className="about-footer">
    Good luck !
    <br />
    <br />
    <a
      href="https://twitter.com/hihuz"
      rel="noopener noreferrer"
      target="_blank">
      @hihuz
    </a>
  </p>
</div>
`;
